src/app/app.component.ts

```diff
@@ -200,6 +203,20 @@ export class AppComponent implements OnInit, AfterViewInit {
   public onHideLoadingIndicator() {
     this.lazyLoading$.next(false);
   }
+
+  private initSmartlook() {
+    if (environment.smartlookKey) {
+      this.store$
+        .pipe(
+          select(selectCurrentUser),
+          first(),
+          filter(user => !allowedEmails.includes(user.email))
```

Comment:
If you access the application without being logged in, `user` can be `null`. We need to use `filter` operator here the same way as in `setUpExternalServicesUserContext()` method.